### PR TITLE
Debian: Add additional package dependency 'ca-certificates', remove 'libgomp1'

### PR DESCRIPTION
**Pull request by mw9**
These proposed changes update the Debian package dependencies as follows:

- Add `ca-certificates`

A dependency on `libio-socket-ssl-perl` (`IO::Socket::SSL`) was added in February 2018 to ensure that the LMS installation is capable of using HTTPS.

However, this is not enough to provide seamless operation. At present, many HTTPS operations will simply fail in the absence of a valid set of root certificates. On a Debian system the `ca-certificates` package provides a collection apparently shipped with Mozilla's browser.

https://packages.debian.org/buster/ca-certificates

The issue came to light when I installed LMS in a minimal Debian installation that did not have the root certificates installed. It was impossible to enter login details for mysqueezebox.com during the first time setup web page, due to HTTPS failures. And there were many other HTTPS failures.

In a nutshell: if we want `IO::Socket::SSL`, then we will want some root certificates.

I've proposed that the package be recorded as a hard dependency because LMS simply won't function satisfactorily without root certificates, and I am not aware of an alternative Debian package.

Another approach would be to leave the installation of root certificates to the user. Both `wget` and `curl` packages take this approach, they _recommend_ the package, without recording a hard dependency. But then they can operate without it.

- Remove `libgomp1`

The LMS bundled Sox binary was updated in August 2018. Amongst other things, OpenMP support was removed from the build, and there is no longer a dependency on the OpenMP library (`libgomp1`).

So removing this dependency as it no longer exists.

### CODE REVIEW

**Comment by mherger** [comment]
Let's give this a try - thanks!

**Comment by mw9** [comment]
I guess if a _'use insecure https'_ option that has been suggested elsewhere were implemented, one might 'demote' `ca-certificates` to a recommendation. Perhaps accompanied by a potentially annoying installer prompt.

I don't know how hard/intricate that would be, I don't have the knowledge of perl's SSL/TLS handling options. I suspect that it may require rather careful testing on a variety of platforms.

And, if done, do you then have LMS check that 'secure' SSL/TLS, as opposed to 'insecure' SSL/TLS is also actually functional, in addition to LMS' current check for SSL/TLS support ?

**Comment by nbeck** [comment]
Unsubscribe
